src/transform/src/join_implementation.rs

```diff
@@ -646,33 +661,36 @@ impl<'a> Orderer<'a> {
                         }
                     });
                     // ... find the equivalence it belongs to ...
-                    let key_equivalence = self
+                    if let Some(key_equivalence) = self
                         .equivalences
                         .iter()
                         .position(|e| e.iter().any(|expr| expr == &k))
-                        .unwrap();
```

Comment:
The unwrap has been changed to `if let Some(key_equivalence) = ...` because this part immediately crashed when I allowed an arrangement on `a+4` when column `a` has been bound.  

Now the crash happens at the render stage.

Reply:
Can you say more about this? There are queries that we can write that now crash Materialize in rendering? Or have always crashed, but haven't been tested before?

Reply:
Nothing crashed before, and nothing will crash AFAIK post PR-merging.

We previously discussed how currently on `main`, the test in `order_input` would allow an arrangement with key `a + b` as long as `a` and `b` are bound. In practice, the code allowing `a+b` as long as `a` and `b` are bound never ran because we exclude all indexes on expression keys from consideration earlier on.

So I took out the code that excluded indexes on expression keys, and I ran a test query (see line 180, join_index.slt) to see what happens if we actually run the code allowing an arrangement with key `a + b` as long as `a` and `b` are bound. The test query immediately crashed at this line, so I fixed it (as you can see above). With the fix, the test query crashes in the render stage.

The crash is avoided in this PR because the test is stricter, requiring that `a + b` be bound itself.

Reply:
In other words, this is crash-proofing for the future of allowing `a+b` as long as `a` and `b` are bound.